packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -21,69 +35,363 @@ import styled from 'styled-components';
+ * ReleaseDetailsLayout
+ * -----------------------------------------------------------------------------------------------*/
+const ReleaseInfoWrapper = styled(Flex)`
+  align-self: stretch;
+  border-bottom-right-radius: ${({ theme }) => theme.borderRadius};
+  border-bottom-left-radius: ${({ theme }) => theme.borderRadius};
+  border-top: 1px solid ${({ theme }) => theme.colors.neutral150};
+`;
 
 const PopoverButton = styled(Flex)`
   align-self: stretch;
 `;
 
 const PencilIcon = styled(Pencil)`
-  width: 16px;
-  height: 16px;
+  width: ${({ theme }) => theme.spaces[4]};
+  height: ${({ theme }) => theme.spaces[4]};
   path {
     fill: ${({ theme }) => theme.colors.neutral600};
   }
 `;
 
 const TrashIcon = styled(Trash)`
-  width: 16px;
-  height: 16px;
+  width: ${({ theme }) => theme.spaces[4]};
+  height: ${({ theme }) => theme.spaces[4]};
   path {
     fill: ${({ theme }) => theme.colors.danger600};
   }
 `;
 
-const ReleaseInfoWrapper = styled(Flex)`
-  align-self: stretch;
-  border-radius: 0 0 4px 4px;
-  border-top: 1px solid ${({ theme }) => theme.colors.neutral150};
-`;
+interface PopoverWrapperProps {
+  onClick?: () => void;
+  children: React.ReactNode;
+}
 
-const ReleaseDetailsPage = () => {
+const PopoverWrapper = ({ onClick, children }: PopoverWrapperProps) => {
+  return (
+    <PopoverButton
+      paddingTop={2}
+      paddingBottom={2}
+      paddingLeft={4}
+      paddingRight={4}
+      alignItems="center"
+      gap={2}
+      as="button"
+      hasRadius
+      onClick={onClick}
+    >
+      {children}
+    </PopoverButton>
+  );
+};
+
+interface ReleaseDetailsLayoutProps {
+  toggleEditReleaseModal: () => void;
+  children: React.ReactNode;
+}
+
+export const ReleaseDetailsLayout = ({
+  toggleEditReleaseModal,
+  children,
+}: ReleaseDetailsLayoutProps) => {
+  const { formatMessage } = useIntl();
   const { releaseId } = useParams<{ releaseId: string }>();
-  const [releaseModalShown, setReleaseModalShown] = React.useState(false);
   const [isPopoverVisible, setIsPopoverVisible] = React.useState(false);
   const moreButtonRef = React.useRef<HTMLButtonElement>(null!);
-  const { formatMessage } = useIntl();
-  const toggleNotification = useNotification();
-  const { formatAPIError } = useAPIErrorHandler();
-  // TODO: get title from the API
-  const title = 'Release title';
+  const { data, isLoading: isLoadingDetails } = useGetReleaseQuery({ id: releaseId });
 
-  const totalEntries = 0; // TODO: replace it with the total number of entries
-  const days = 0; // TODO: replace it with the number of days since the release was created
-  const createdBy = 'John Doe'; // TODO: replace it with the name of the user who created the release
+  const release = data?.data;
+
+  const title = release?.name;
+  const totalEntries = release?.actions?.meta?.count || 0;
+
+  const createdBy = `${release?.createdBy?.firstname} ${release?.createdBy?.lastname}` || '';
```

Comment:
Same sort of thing as the other component. I would handle the case for no release before render and then access all this stuff after that. Personally I would just access it directly in the success return. You will have to determine the best way to do that but for example if you do

`if (!release) return null` before your return the UI then you are sure to have everything a release has and you don't need these fallbacks.